src/dvclive/live.py

```diff
@@ -379,6 +397,17 @@ def step(self, value: int) -> None:
         self._step = value
         logger.debug(f"Step: {self.step}")
 
+    @property
+    def monitor_cpu(self) -> Optional[MonitorCPU]:
+        return self._monitor_cpu or None
+
+    @monitor_cpu.setter
+    def monitor_cpu(self, monitor_cpu: MonitorCPU) -> None:
+        if self._monitor_cpu is not None:
+            self._monitor_cpu.end()
+        self._monitor_cpu = monitor_cpu
+        self._monitor_cpu(self)
+
```

Comment:
Is it necessary to have this and to have similar logic in init and end? Why is it here for monitor_cpu but there's nothing similar in the gpu pr?

Reply:
I updated the GPU PR, and I'll keep updating it based on this discussion.

What do you have in mind? If we don't call `end` the thread will continue. Do you want to keep the same instance?
Something like:
```
with Live(monitor_system=True) as live:
    live.monitor_cpu.interval = 0.1
```
?

Reply:
I should definitely make the `end` method private. I don't want the user to use that for now. 

I'm not sure I get you on the first question: "We are starting and ending the thread from the internal self._monitor_cpu already, right?". 
Correct me if I'm wrong, but if the user creates a `Live` object it will assign an instance of CPUMonitor (call "A" for the example). Then it creates another `CPUMonitor` with custom parameters (instance called "B"), and give it to the `Live` object with `live.cpu_monitor = CPUMonitor(...)`. 
If I don't end the instance A, if will continue to run in the background no? This is why I called the `end` method here.

Reply:
> Then it creates another `CPUMonitor` with custom parameters (instance called "B"), and give it to the `Live` object with `live.cpu_monitor = CPUMonitor(...)`.

I didn't understand that you want to expose this option to users. It means we will also have to document the whole CPUMonitor class, which breaks the current docs structure of the whole API being inside live.py. It also makes me wonder whether it makes sense to have separate CPUMonitor and GPUMonitor since customizing will likely mean needing to override both. It sounds nice to give all that flexibility, but not sure if it's needed yet/might be better in a separate PR.

Reply:
> I didn't understand that you want to expose this option to users.

Initially, we could have used a hidden object (and the user could not change the interval, and it was ok). But since we added the disk usage statistics, we **need** user inputs. There is no way for us to guess which partition the user wants to monitor. We could guess "/", but there is a high chance the data/images are not located here.

If we want to hide how the CPU monitoring is done to the user, we need to remove the disk usage statistics. 

I agree that the doc organisation could be a problem, even tho we are not forced to do the doc in full details now. The main behavior is to control the system metrics with a boolean value (True => save my system metrics, False => don't). The option of changing the CPUMonitor object can be explained later on if we want to :)

Reply:
> It also makes me wonder whether it makes sense to have separate CPUMonitor and GPUMonitor since customizing will likely mean needing to override both. It sounds nice to give all that flexibility, but not sure if it's needed

Yeap really good point. I agree on the flexibility boost, but I also agree that it is annoying for the user. There will be lots of import and code to change just the `interval` value. 

I can group them together and make them use the same thread, so `interval`, `num_samples`, and `plot` would be shared. What do you think?

Reply:
Okay, all your points make sense, esp. if we agree we don't need to document all options immediately, and sorry for the confusion. I'll resolve and leave up to you what you think is best for the public property, including/excluding disk usage, and combining/separating cpu/gpu threads.